pkg/telemetry/telemetry_test.go

```diff
@@ -0,0 +1,194 @@
+package telemetry
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/armon/go-metrics"
+	"github.com/google/uuid"
+	"github.com/square/spiffe-aws-assume-role/internal/mocks"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+var anyLabels string = mock.Anything
+
+func TestInstrumentCalls(t *testing.T) {
+	metricSink := mocks.MetricSink{}
+	allowAllCalls(&metricSink)
+
+	telemetry, err := NewTelemetryForSink(&metricSink)
+	require.NoError(t, err)
+
+	emitMetrics := telemetry.Instrument([]string{"foo", "bar"}, &err)
+	emitMetrics()
+
+	metricSink.AssertCalled(t, "IncrCounterWithLabels",
+		[]string{"spiffe-aws-assume-role", "foo", "bar", "Calls"},
+		float32(1),
+		anyLabels)
```

Comment:
The "service name" is "spiffe-aws-assume-role". It gets prepended as the first key for every metric that's emitted. Yay/nay?  I have reason to believe that dashes are not allowed in keys. Can also covert to a label.

Reply:
Is there a way to identify that the metrics are associated with `spiffe-aws-assume-role` otherwise? Feels a bit weird that it gets prepended, despite it not being in the `key` slice when calling `Instrument` and not being appended as a label.

Maybe better to convert to a label?

Reply:
Yeah, I honestly don't know how the choice affects the end result functionally, it might be a wash. I'll convert it to a label.